src/host_abi.zig

```diff
@@ -0,0 +1,116 @@
+/// All Roc functions that are exposed to the host take 1 argument and return void.
+/// The 1 argument is a pointer to one of these structs, which includes an address
+/// that the Roc call will write the return value into. This design makes Roc's ABI
+/// very simple; the calling convention is just "pass a single pointer".
+pub fn RocCall(
+    comptime Arg: type,
+    comptime Ret: type,
+    comptime CallEnv: type,
```

Comment:
These types will all be `anyopaque` for an actual host, but we can get value out of them being type parameters when using this inside the compiler (e.g. for compile-time evaluation of constants).

Reply:
Would this work in practice? Cause we won't know the types at zig comptime...only at roc comptime.

Reply:
Hm that's a good point...maybe these should just be type aliases to make them be self-documenting.

Reply:
Still useful for `CallEnv` and `HostFns`, since those are known at comptime (when evaluating constants).